### PR TITLE
Add basic initial_context() support

### PR DESCRIPTION
**Pull request by dburgener**
There are a few TODOs left, but this should get basic support in the happy path in.

### CODE REVIEW

File: src/functions.rs [hunk 1]
```diff
@@ -961,6 +961,75 @@ fn call_to_fsc_rules<'a>(
...
+    context: &BlockContext<'_>,
+    file: &SimpleFile<String, String>,
+) -> Result<Vec<Sid<'a>>, CascadeErrors> {
+    if context.in_function_block() {
+        return Err(CascadeErrors::from(
+            ErrorItem::make_compile_or_internal_error(
+                "initial_context() calls are not valid in functions",
+                Some(file),
+                c.name.get_range(),
+                "Remove this call",
```
Comment (matt-sheets) [inline src/functions.rs:977]: add "or move to global scope"?

File: src/functions.rs [hunk 2]
```diff
@@ -961,6 +961,75 @@ fn call_to_fsc_rules<'a>(
...
+        .get_name_or_string(context)?;
+    let sid_context = match Context::try_from(context_str.to_string()) {
+        Ok(c) => c,
+        Err(_) => {
+            return Err(CascadeErrors::from(
+                ErrorItem::make_compile_or_internal_error(
+                    "Invalid context",
+                    Some(file),
+                    context_str.get_range(),
+                    "Cannot parse this into a context",
```
Comment (matt-sheets) [inline src/functions.rs:1024]: I looked around a bit on this one and we do this exact thing with Context::try_from in a few places.  But I feel like we should be passing up some info in the error we catch here that will help the user out a bit more.  Things like invalid user, role, type, mls at the very least.

I don't vote we fix it in this PR but should go on the TODO pile.

File: src/functions.rs (hunk 1 above)
Comment (dburgener) [reply src/functions.rs:977]: I thought about that.  The problem is that the most idiomatic thing for a user to do would be to put it inside a type definition.  Global scope is fine of course, but I would expect the common case to be:

```
resource kernel_sid {
    initial_context("kernel", this);
}
```

Maybe this rewrite would be more helpful?  "You may want to place this call directly inside a type definition"

File: src/functions.rs (hunk 2 above)
Comment (dburgener) [reply src/functions.rs:1024]: #171

File: src/compile.rs [hunk 3]
```diff
@@ -1070,7 +1086,20 @@ pub fn get_reduced_infos(
     )?
     .inner(&mut warnings);
 
-    validate_rules(&new_policy_rules)?;
+    // TODO: This is a hack to work around the fact that SIDs contain Contexts, which *might*
```
[the remark this answers is not in the document]
Comment (dburgener) [reply src/compile.rs:1089]: #172

File: src/functions.rs (hunk 1 above)
Comment (matt-sheets) [reply src/functions.rs:977]: I like that better.

**Pull request merged**